SFS_monochromatic/driving_functions_mono/driving_function_mono_wfs_pw.m

```diff
@@ -1,143 +1,160 @@
+xref = conf.xref;
+c = conf.c;
+dimension = conf.dimension;
+driving_functions = conf.driving_functions;
+
+
+%% ===== Computation ====================================================
+% Calculate the driving function in time-frequency domain
+
+% Frequency
+omega = 2*pi*f;
+
+
+if strcmp('2D',dimension) || strcmp('3D',dimension)
+
+    % === 2- or 3-Dimensional ============================================
+
+    if strcmp('default',driving_functions)
+        % --- SFS Toolbox ------------------------------------------------
+        % D using a plane wave as source model
+        %
+        %              i w
+        % D(x0,w) =  2 --- nk nx0  e^(-i w/c nk x0)
+        %               c
+        %
+        % See http://sfstoolbox.org/#equation-D.wfs.pw
+        %
+        D = 2.*1i.*omega./c .* vector_product(nk,nx0,2) ...
+            .* exp(-1i.*omega./c.*vector_product(nk,x0,2));
+        %
+    else
+        error(['%s: %s, this type of driving function is not implemented ', ...
+            'for a plane wave.'],upper(mfilename),driving_functions);
+    end
+
+
+elseif strcmp('2.5D',dimension)
+
+    % === 2.5-Dimensional ================================================
+
+    % Reference point
+    xref = repmat(xref,[size(x0,1) 1]);
+    switch driving_functions
+    case {'default', 'reference_point'}
+        % Driving function with only one stationary phase approximation, i.e.
+        % reference to one point in field
+        %        ______________
+        % g0 = \| 2pi |xref-x0|
+        %
+        g0 = sqrt( 2*pi*vector_norm(xref-x0,2) );
+        %
+        % D_2.5D using a plane wave as source model
+        %                             ___
+        %                            |i w|
+        % D_2.5D(x0,w) = g0 nk nx0 _ |---  e^(-i w/c nk x0)
+        %                           \| c
+        %
+        % See http://sfstoolbox.org/#equation-D.wfs.pw.2.5D
+        %
+        D = 2.*g0 .* vector_product(nk,nx0,2) .* sqrt(1i.*omega./c) ...
+            .* exp(-1i.*omega./c.*vector_product(nk,x0,2));
+        %
+    case {'reference_line'}
+        % Driving function with two stationary phase approximations,
+        % reference to a line parallel to a LINEAR secondary source distribution
+        %
+        % Distance ref-line to linear ssd
+        dref = vector_product(xref-x0,nx0,2);
+        %
+        % 2.5D correction factor
+        %        ____________
+        % g0 = \| 2pi * d_ref
+        %
+        g0 = sqrt(2.*pi.*dref);
+        %
+        % D_2.5D using a plane wave as source model
+        %                       ___
+        %                      |i w|
+        % D_2.5D(x0,w) = 2g0 _ |---  e^(-i w/c nk x0)
+        %                     \| c
+        %
+        % See Schultz (2016), eq. (2.170)
+        %
+        D = 2.*g0 .* sqrt(1i*omega/c) .* sqrt(vector_product(xref-x0,nx0,2)) ...
+            .* exp(-1i.*omega./c.*vector_product(nk,x0,2));

```

Comment:
x0 vs. nx0 ???

I've this implemented and checked for the thesis:

rather needs to be replaced by ``sqrt(vector_product(nk,nx0,2))

Can somebody doublecheck?


On 15.06.2016 17:11, Fiete Winter wrote:
>> +        % 2.5D correction factor
>> +        %        ____________
>> +        % g0 = \| 2pi * d_ref
>> +        %
>> +        g0 = sqrt(2.*pi.*dref);
>> +        %
>> +        % D_2.5D using a plane wave as source model
>> +        %                       ___
>> +        %                      |i w|
>> +        % D_2.5D(x0,w) = 2g0 _ |---  e^(-i w/c nk x0)
>> +        %                     \| c
>> +        %
>> +        % See Schultz (2016), eq. (2.170)
>> +        %
>> +        D = 2.*g0 .* sqrt(1i*omega/c) .* sqrt(vector_product(xref-x0,nx0,2)) ...
>> +            .* exp(-1i.*omega./c.*vector_product(nk,x0,2));
>
> Not the formula from Schultz (2016), eq. (2.170),  ``sqrt(vector_product(xref-x0,nx0,2))`` needs to be replaced by ``sqrt(vector_product(nk,x0,2))``
>
> ---
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/sfstoolbox/sfs/pull/37/files/673b01f005725f8d2d83264fcf5f2438445a50f1#r67181009
>